### PR TITLE
#386: Added environment-specific configurations.

### PR DESCRIPTION
**Pull request by akara**
Thanks for your pull request.  Please review the following guidelines.

- [X] Title includes issue id.
- [X] Description of the change added.
- [X] Commits are squashed.
- [X] Tests added.
- [X] Documentation added/updated.
- [X] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by anilgursel** [comment]
If you already have an environment resolution mechanism (by using `EnvironmentResolver`), then this feature just adds more confusion.  IMO, we should not muddy http client by hardcoding any environment specific logic in it.  A squbs user can define an environment specific configuration by just reading environment specific sections of `application.conf`.

**Comment by akara** [comment]
I tripped on this when updating internal stuff (please see internal PR) and found it as a feature regression. That's why the issue and the relatively easy fix.

The `EnvironmentResolver` gives you the environment, but no configuration with it. So to achieve the same, the user will need to resolve the environment manually, then read the config for the desired client and environment from a user-specified location in the config, and finally pass the config to the `ClientFlow` upon creation. Don't you think that is harder to manage and support than having a well known location for environment-specific client configuration?

We better discuss face-to-face. Thanks!

**Comment by anilgursel** [comment]
I agree, let's discuss.

File: docs/httpclient.md [hunk 1]
```diff
@@ -192,6 +192,26 @@ sample {
...
+    akka.http.host-connection-pool {
+      max-connections = 4
+    }    
+  }
+}
+```
+
+In this case if the `ClientFlow` is created for the `QA` environment, the settings `max-connections = 4` would apply instead of `max-connections = 10` as specified for all other environments.
+
 ### Pipeline
```
Comment (DaJoker78) [inline docs/httpclient.md]: Ithoughtyouknew

Comment (DaJoker78) [inline docs/httpclient.md]: Tre4

Comment (DaJoker78) [inline docs/httpclient.md]: Shaggy34

Comment (DaJoker78) [inline docs/httpclient.md]: chadmichaellheureux

Comment (DaJoker78) [inline docs/httpclient.md]: World Inc

Comment (DaJoker78) [inline docs/httpclient.md]: Me

Comment (DaJoker78) [inline docs/httpclient.md]: HappyNes

**Pull request closed without merging**